iosxr_iso2vbox.py

```diff
@@ -565,41 +547,42 @@ def main(argv):
     logger.debug('Boot order DVD second')
     run(['VBoxManage', 'modifyvm', vmname, '--boot2', 'dvd'])
 
+    return vbox
+
+def launch_and_configure_vbox_vm(vmname, box_dir, verbose, pause_to_debug):
+    """Start the VM, add XR and Linux configs, and power it off when done."""
     # Start the VM for installation of ISO - must be started as a sub process
     logger.debug('Starting VM...')
     start_process(['VBoxHeadless', '--startvm', vmname])
 
+    elapsed_time = 0
     while True:
-        vms_list = run(['VBoxManage', 'showvminfo', vmname])
-        if 'running (since' in vms_list:
+        vms_list_running = run(['VBoxManage', 'list', 'runningvms'])
+        if re.search('"' + vmname + '"', vms_list_running):
             logger.debug('Successfully started to boot VM disk image')
             break
-        else:
-            logger.warning('Failed to install VM disk image\n')
+        elif elapsed_time < 600:
+            logger.warning("VM is not yet running after %d seconds; "
+                           "sleep 5 seconds and retry", elapsed_time)
             time.sleep(5)
+            elapsed_time = elapsed_time + 5
             continue
+        else:
+            # Dump verbose output in case it helps...
+            run(['VBoxManage', 'showvminfo', vmname])
+            raise AbortScriptException(
+                "VM still not running after {0} seconds!"
+                .format(elapsed_time))
 
     # Configure IOS XR and IOS XR Linux
-    configure_xr(args.verbose)
+    configure_xr(verbose)
 
     # Good place to stop and take a look if --debug was entered
-    if args.debug:
+    if pause_to_debug:
         pause_to_debug()
```

Comment:
So this would become something like `maybe_pause_to_debug(args.debug)`? Could do but I'm fine with the current approach as well.